app/src/main/java/com/capstone/weatherapp/WeatherRepository.kt

```diff
@@ -26,23 +26,40 @@ class WeatherRepository(private val db: WeatherDatabase) {
 
     suspend fun refreshCityList() {
         withContext(Dispatchers.IO) {
+            _cityListData.postValue(State.Loading)
             try {
-                val cityList = WeatherApiClient.retrofitService.getCities(CITIES, APIKEY, UNITS)
-                db.cityDao().insert(cityList.convertToCityCacheList())
-            } catch (e: Exception) {}
-            _cityListData.postValue(getNetworkCityList())
+                val cityListResponse = WeatherApiClient.retrofitService.getCities(CITIES, APIKEY, UNITS)
+                if (cityListResponse.isSuccessful) {
+                    val response = cityListResponse.body()!!
+                    _cityListData.postValue(State.Success(response.list))
+                    db.cityDao().insert(response.convertToCityCacheList())
+                }
+                else {
+                    _cityListData.postValue(State.Success(getNetworkCityList()))
+                }
+            } catch (e: Exception) {
+                _cityListData.postValue(State.Error(e.toString()))
+            }
         }
```

Comment:
so, after the initial try catch block, we need something else. The first block should be used to fetch api data and insert it into the db. Then, we need another try catch block to fetch the data from the db. give it a try yourself before I help too much